### PR TITLE
Adds configurable settings

### PR DESCRIPTION
**Pull request by sam-packer**
- Adds customizing the table and features
- Adds choosing where you want the neural network files outputted
- Adds configuring the layers inside the neural network
- Adds configuring the embedding dimensions
- Allows to configure the training hyperparameters
- Allows you to change evaluation metrics
- Moves MLFlow from system environment to config file
- Allows configuring Shibboleth deployment from config on top of CLI
- Allows configuring the API options

### CODE REVIEW

File: nn_scripts/explain_nn.py
Comment (Michaelsaju1) [inline nn_scripts/explain_nn.py]: tsk tsk. I recommend replacing the logging warning with a professional message as that is not very kind.

File: config.yml
[the remark this answers is not in the document]
Comment (Telesphoreo) [reply config.yml]: There aren't. FEATURE_COLUMNS is no longer a hard coded list in globals.py. The variable may exist in some places and be inline and others but it always does reference the config.yml file.

File: helpers/verify_config_usage.py
Comment (Michaelsaju1) [inline helpers/verify_config_usage.py]: I recommend adding validation for missing required keys or type mismatches (such as learning rate must be a float).

File: web/run_api.py
Comment (Michaelsaju1) [inline web/run_api.py]: I recommend changing api.threats to api.threads because it is mispelles and Gunicorn probably wouldn't pick up the intended concurrency.

File: nn_scripts/explain_nn.py
Comment (sam-packer) [reply nn_scripts/explain_nn.py]: But I like being mean.

File: helpers/verify_config_usage.py
Comment (sam-packer) [reply helpers/verify_config_usage.py]: This file is actually meant for developer usage and to ensure that all the keys that exist are being used. You raise an EXCELLENT point broadly though. I should ensure that the types match what they are expected to be. Or, add defaults to every config variable and warn the user "yours didn't work, using default"

File: web/run_api.py
Comment (sam-packer) [reply web/run_api.py]: Thanks for catching this!
